### DISCUSSION TITLE
Default configuration values

### DISCUSSION THREAD

**Discussion by gsketefian** [dtcenter/METplus#1737] [Configuration] [answered]
Hi,

In the MET/METplus installation on Hera (base at `/contrib/met/10.1.1`), I found two sets of default configuration files:

1) `share/met/config`
2) `met-10.1.1/data/config`

I’m wondering what the order is in which these are read.  Also, is there another layer of default values when using METplus wrappers (which is what I’m using)?  

Thanks.

**Comment by georgemccabe** [comment, marked as answer]
Hi @gsketefian,

> I’m wondering what the order is in which these are read.

The files in share/met/config are read by the MET tools and the files in met-X.Y.Z/data/config are not. The met-10.1.1/data/config files are from the MET repository and are copied into share/met/config upon installation. The installation area and the source code may not always be in the same location, so the files in met-10.1.1 may not always be present in the installation area.

> Also, is there another layer of default values when using METplus wrappers (which is what I’m using)?

When a MET tool is run, the default values found in the share/met/config files are read first. Next, any user provided config file is read. The METplus wrappers have a set of ["wrapped" MET config files](https://github.com/dtcenter/METplus/tree/main_v4.1/parm/met_config) that contain environment variables that are set by the wrappers to override the defaults from share/met/config. The wrappers can be configured to use a different wrapped MET config file if desired, but this approach is discouraged because user-provided MET config files may not contain all of the environment variables that are set in the wrappers.

Starting in METplus 4.0.0, the logic was changed to the set environment variables used in the wrapped MET config files that include the full "name = value;" or an empty string. When the env var is set to an empty string, then the default value from the share/met/config file is used. This prevents the need to keep the default values that are hard-coded in the wrapped MET config files in sync with the defaults in share/met/config. Unfortunately this transition brought to light that there were hard-coded default values in the wrapped files that did not match the defaults in MET. Eventually we would like to update all of the wrapped MET config files to not contain any hard-coded defaults, but this is not currently the case. Comparing the default values set in two MET config files would uncover any mismatches.
